lib/runner.js

```diff
@@ -123,16 +123,22 @@ module.exports = Runner;
  * @class
  * @param {Suite} suite Root suite
  * @param {boolean} [delay] Whether to delay execution of root suite until ready.
+ * @param {boolean} [cleanReferencesAfterRun] Whether to clean references to test fns and hooks when a suite is done.
  */
-function Runner(suite, delay) {
+function Runner(suite, delay, cleanReferencesAfterRun) {
```

Comment:
we're going to be collecting options here (#4245 does it as well).  Let's change the function signature, but in a non-breaking way:

```js
/**
 * Initialize a `Runner` at the Root {@link Suite}, which represents a hierarchy of {@link Suite|Suites} and {@link Test|Tests}.
 *
 * @extends external:EventEmitter
 * @public
 * @class
 * @param {Suite} suite - Root suite
 * @param {Object|boolean} [opts] - Options. If `boolean`, whether or not to delay execution of root suite until ready (for backwards compatibility).
 * @param {boolean} [opts.delay] - Whether to delay execution of root suite until ready.
 * @param {boolean} [opts.cleanReferencesAfterRun] - Whether to clean references to test fns and hooks when a suite is done.
 */
function Runner(suite, opts) {
  var delay;
  if (typeof opts === 'boolean') {
    delay = opts;
    opts = {};
  }
  // etc
```

Reply:
Good point. Will fix.